### DISCUSSION TITLE
Welcome to SwissRailwayClock Discussions!

### DISCUSSION THREAD

**Discussion by ahuggel** [ahuggel/SwissRailwayClock#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by midget1234** [comment]
I just love it and would really love to have this on my Fenix 5. Can you help me a little bit with how to Install it ? 
I understand it is not available in the IQ shop probably due to IP issues. But I would not know how to install it.. Can you provide some basic instructions ? Thanks!

**Reply by ahuggel** [reply]
You'd have to find a tutorial with the steps to install and configure a development environment, I'm sure there are plenty out there. Essentially, you need to install Visual Studio Code (VSC), the VSC Monkey C extension, the Prettier Monkey C VSC extension, Garmin Connect IQ SDK and Git. Then you can pull the source code for the watchface from here, compile it for your watch and "sideload" it.

Exactly what watch model do you have (5 Plus, 5S Plus or 5X Plus)? I'd love to hear feedback if/how it works on an actual Fenix 5 series watch (as opposed to just the simulator). So if you like, I can compile and upload the program here, then you'll just need to load it onto the watch.

**Comment by midget1234** [comment]
Fenix 5 Plus.That process sounds a bit complicated to me 😉. All help is welcome here. More than happy to test functionality on 5plus and also donate to the development.Thx, RuudVerstuurd vanaf mijn iPhoneOp 12 jan 2024 om 08:45 heeft Andreas Huggel ***@***.***> het volgende geschreven:﻿
You'd have to find a tutorial with the steps to install and configure a development environment, I'm sure there are plenty out there. Essentially, you need to install Visual Studio Code (VSC), the VSC Monkey C extension, the Prettier Monkey C VSC extension, Garmin Connect IQ SDK and Git. Then you can pull the source code for the watchface from here, compile it for your watch and "sideload" it.
Exactly what watch model do you have (5 Plus, 5S Plus or 5X Plus)? I'd love to hear feedback if/how it works on an actual Fenix 5 series watch (as opposed to just the simulator). So if you like, I can compile and upload the program here, then you'll just need to load it onto the watch.

**Reply by ahuggel** [reply]
I've uploaded a binary, you can get it from GitHub [here](https://github.com/ahuggel/SwissRailwayClock/releases/tag/v1). To load it onto your watch using a Windows PC, connect the watch with the USB cable and copy the binary to the `GARMIN\Apps` folder on the watch. If you're using a Mac, search for instructions to "sideload a watchface to a Garmin watch" to find out how it works there.
Then you can select the newly installed watchface via the "Watch Face" menu of your watch as usual, and click "Apply" to confirm. To access the watchface settings menu, choose "Customize" instead of "Apply". 
To uninstall it, use the Garmin Connect IQ app rather than deleting it from the folder directly.
I'd appreciate if you played with the various settings and let me know if you encounter any issues in the process. Thanks!

**Reply by marcos9989** [reply]
So happy to have this back on my Fenix 6s.  Working like a charm.  Thanks for compiling!

**Reply by ahuggel** [reply]
Indeed, the program compiled for the Fenix 6s is identical to the binary for 5 Plus. Interesting..

**Comment by midget1234** [comment]
Installation done!Had to revert to Windows as I could not figure it out with Mac.I will play and test with the options/custimization.🙏Verstuurd vanaf mijn iPhoneOp 12 jan 2024 om 10:02 heeft Andreas Huggel ***@***.***> het volgende geschreven:﻿
I've uploaded a binary, you can get it from GitHub here. To load it onto your watch using a Windows PC, connect the watch with the USB cable and copy the binary to the GARMIN\Apps folder on the watch. If you're using a Mac, search for instructions to "sideload a watchface to a Garmin watch" to find out how it works there.
Then you can select the newly installed watchface via the "Watch Face" menu of your watch as usual, and click "Apply" to confirm. To access the watchface settings menu, choose "Customize" instead of "Apply".
To uninstall it, use the Garmin Connect IQ app rather than deleting it from the folder directly.
I'd appreciate if you played with the various settings and let me know if you encounter any issues in the process. Thanks!
